client/src/store/actions/webSocketActions.js

```diff
@@ -0,0 +1,26 @@
+import { createAction } from 'redux-actions';
+
+const types = {
+    connect: 'WS_CONNECT',
+    close: 'WS_CLOSE',
+    send: 'WS_SEND',
+    received: 'WS_DATA_RECEIVED'
+}
+
+const connectWebSocket = createAction(types.connect);
+const closeWebSocket = createAction(types.close);
+const sendWebSocket = createAction(types.send);
+const dataRecived = createAction(types.received);
+const connectedWebSocket = createAction('WS_CONNECTED');
```

Comment:
why did you stop the types here ? and why are the types not in a config file ?

Reply:
so after the long post i wrote about EDIT_USER, here I didn't do it all the way, there is a reason for that too (not a great one but still).
all those are actions that should be catch by the middleware (webSocketMiddleware), for them to catch an action they need to know the type of the action, so im exporting all the types, the middleware needs.
The reducers on the other hand don't need the actual string (the type ) the can catch the actions by the action creator (userReducer.js line 8 for example)